sk0-back/src/users/services/users.service.ts

```diff
@@ -19,12 +19,11 @@ export class UsersService {
   ) {}
 
   async create(user: CreateUserDto): Promise<User> {
-    const createdUser = this.usersRepository.create({ ...user });
     const userByEmail = await this.findOneByEmail(user.email);
     if (userByEmail) {
       throw new UserExistException("Exists", 400);
     }
-
+    const createdUser = this.usersRepository.create({ ...user });
```

Comment:
Are we creating and re-saving the entity? Is it necessary? Isn't it just an extra database transaction?

Reply:
We did it because it was the typeorm recommended way.